src/Solver/Interoperability.cpp

```diff
@@ -555,8 +631,50 @@ void seissol::Interoperability::setMaterial(int i_meshId, int i_side, double* i_
     assert(side < 4);
     material = &m_ltsLut.lookup(m_lts->material, i_meshId - 1).neighbor[side];
   }
+  //TODO(Lukas)
+  //Use placement new because pointer to virtual function table gets overwritten by 0 during init.
+#if defined USE_ANISOTROPIC
+  new(material) seissol::model::AnisotropicMaterial;
+#elif defined USE_VISCOELASTIC || defined USE_VISCOELASTIC2
+  new(material) seissol::model::ViscoElasticMaterial;
+#else 
+  new(material) seissol::model::ElasticMaterial;
+#endif
+
+  switch(material->getMaterialType()) {
+    case seissol::model::MaterialType::anisotropic: {
+      seissol::model::AnisotropicMaterial* am = dynamic_cast<seissol::model::AnisotropicMaterial*>(material);
+      seissol::model::setMaterial(i_materialVal, i_numMaterialVals, am);
+      break;
+    }
+    case seissol::model::MaterialType::elastic: {
+      seissol::model::ElasticMaterial* em = dynamic_cast<seissol::model::ElasticMaterial*>(material);
+      seissol::model::setMaterial(i_materialVal, i_numMaterialVals, em);
+      break;
+    }                                          
+    case seissol::model::MaterialType::viscoelastic: {
+      seissol::model::ViscoElasticMaterial* vm = dynamic_cast<seissol::model::ViscoElasticMaterial*>(material);
+      seissol::model::setMaterial(i_materialVal, i_numMaterialVals, vm);
+      break;
+    }                                          
+  }
+}
```

Comment:
While I think that the new Material parameter structure is nice, I really dislike the above (and similar statements). These ifdef and switch statements are now scattered throughout the whole code. It is going to be really hard to track all of these down when other PDEs are added in the future.